### ISSUE TITLE
Getting duplicates symbols after installing expo-notifications in a bare project (ios)

### ISSUE THREAD

**Issue by Speedy1991** [expo/expo#8745]
Getting duplicates symbols after installing expo-notifications in a bare project

`yarn add expo-notifications && cd ios && pod install`

Deploying the app with xcode to a local device leads to duplicate symbols. 

I already checked if there is some manual linking going on, but Libraries are clean. Linking is only against pods.

```
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._reject' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._userNotificationPermissionRequester' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_CLASS_$_EXRemoteNotificationPermissionRequester' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_METACLASS_$_EXRemoteNotificationPermissionRequester' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._remoteNotificationsRegistrationIsPending' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._resolve' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._methodQueue' in:
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
  /Users/AB/Library/Developer/Xcode/DerivedData/kinikana-gwnqyjnevydhebahxkjqtmrmushg/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
```

Targets:
- IOS
- Android (runs fine)

Versions:
```
"expo-notifications": "0.3.1",
"expo-permissions": "8.1.0",
"react-native-unimodules": "0.9.1",
```

I don't know how, but it looks like `libEXNotifications` is colliding with `libEXPermissions`?

**Comment by andyhmltn** [comment]
Also experiencing the same error:

```
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._reject' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._userNotificationPermissionRequester' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_CLASS_$_EXRemoteNotificationPermissionRequester' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_METACLASS_$_EXRemoteNotificationPermissionRequester' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._remoteNotificationsRegistrationIsPending' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._resolve' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
duplicate symbol '_OBJC_IVAR_$_EXRemoteNotificationPermissionRequester._methodQueue' in:
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXNotifications/libEXNotifications.a(EXRemoteNotificationPermissionRequester.o)
    /Users/andy/Library/Developer/Xcode/DerivedData/Estuary-gbtnkqqbmgycdugwuqbavfzcieka/Build/Products/Debug-iphoneos/EXPermissions/libEXPermissions.a(EXRemoteNotificationPermissionRequester.o)
ld: 7 duplicate symbols for architecture arm64
```

**Comment by andyhmltn** [comment]
FYI @Speedy1991 - I was able to successfully build without this error by downgrading to `0.1.4`

**Comment by EQuimper** [comment]
@Speedy1991 look like you are right. Did get the same issue and decide to exclude expo-permissions and I was able to build. For now I will use react-native-permissions

**Comment by Andrea-Arguello** [comment]
Getting this same issue. Will try @andyhmltn's and @EQuimper's solutions.

**Comment by brentvatne** [comment]
can you show me the output of running `pod install` in the ios directory?

**Comment by andyhmltn** [comment]
For my solution, I also removed `expo-permission`. It seems as if `expo-notifications` has some methods for requesting permissions so this worked for me

**Comment by Speedy1991** [comment]
> can you show me the output of running `pod install` in the ios directory?

```
pod install
Adding a custom script phase for Pod RNFBApp: [RNFB] Core Configuration
Detected React Native module pods for RNCAsyncStorage, RNCMaskedView, RNDateTimePicker, RNDeviceInfo, RNFBAnalytics, RNFBApp, RNFBInAppMessaging, RNFBPerf, RNGestureHandler, RNReanimated, RNSVG, RNScreens, RNSentry, react-native-safe-area-context, and rn-fetch-blob
Installing unimodules:
 expo-application@2.1.1 from ../node_modules/expo-application/ios
 expo-av@8.1.0 from ../node_modules/expo-av/ios
 expo-constants@9.0.0 from ../node_modules/expo-constants/ios
 expo-file-system@8.1.0 from ../node_modules/expo-file-system/ios
 expo-font@7.0.0 from ../node_modules/expo-font/ios
 expo-image-loader@1.0.1 from ../node_modules/expo-image-loader/ios
 expo-in-app-purchases@8.1.0 from ../node_modules/expo-in-app-purchases/ios
 expo-keep-awake@8.1.0 from ../node_modules/expo-keep-awake/ios
 expo-linear-gradient@7.0.0 from ../node_modules/expo-linear-gradient/ios
 expo-localization@8.1.0 from ../node_modules/expo-localization/ios
 expo-notifications@0.1.7 from ../node_modules/expo-notifications/ios
 expo-permissions@8.1.0 from ../node_modules/expo-permissions/ios
 expo-web-browser@8.2.0 from ../node_modules/expo-web-browser/ios
 unimodules-app-loader@1.0.2 from ../node_modules/unimodules-app-loader/ios
 unimodules-barcode-scanner-interface@5.1.0 from ../node_modules/unimodules-barcode-scanner-interface/ios
 unimodules-camera-interface@5.1.0 from ../node_modules/unimodules-camera-interface/ios
 unimodules-constants-interface@5.1.0 from ../node_modules/unimodules-constants-interface/ios
 unimodules-core@5.1.2 from ../node_modules/@unimodules/core/ios
 unimodules-face-detector-interface@5.1.0 from ../node_modules/unimodules-face-detector-interface/ios
 unimodules-file-system-interface@5.1.0 from ../node_modules/unimodules-file-system-interface/ios
 unimodules-font-interface@5.1.0 from ../node_modules/unimodules-font-interface/ios
 unimodules-image-loader-interface@5.1.0 from ../node_modules/unimodules-image-loader-interface/ios
 unimodules-permissions-interface@5.1.0 from ../node_modules/unimodules-permissions-interface/ios
 unimodules-react-native-adapter@5.2.0 from ../node_modules/@unimodules/react-native-adapter/ios
 unimodules-sensors-interface@5.1.0 from ../node_modules/unimodules-sensors-interface/ios
 unimodules-task-manager-interface@5.1.0 from ../node_modules/unimodules-task-manager-interface/ios
Analyzing dependencies
Downloading dependencies
Generating Pods project
Integrating client project
Pod installation complete! There are 71 dependencies from the Podfile and 94 total pods installed.
```

I can confirm: Removing `expo-permissions` is working, but not really a fix IMO

**Comment by samehgirgis** [comment]
Speedy 1991

**Comment by Speedy1991** [comment]
Downgrading to 0.1.7 causes an other problem: https://github.com/expo/expo/issues/8084#issuecomment-638332660

**Comment by EQuimper** [comment]
@Speedy1991 make sure to clear your pods. When i did remove it I did make sure to delete my pods folder and Podfile.lock

**Comment by sjchmiela** [comment]
Hey all, sorry for introducing this bug. For those interested, it has been introduced in https://github.com/expo/expo/pull/8486 (you can read the discussion underneath _merge_ event).

Good news is the fix has just landed on `master` and I also published it as `expo-notifications@0.3.2`, so go get the upgrade while it's hot!
